aspnet/fundamentals/configuration.rst

```diff
@@ -49,36 +48,36 @@ The settings required by your application and the mechanism used to specify thos
 
-.. note:: To override nested keys through environment variables in shells that don't support ``:`` in variable names replace them with ``__`` (double underscore).
+.. note:: To override nested keys through environment variables in shells that don't support ``:`` in variable names, replace them with ``__`` (double underscore).
 
 It can be useful to have environment-specific configuration files. This can be achieved using the following:
 
 .. literalinclude:: /../common/samples/WebApplication1/src/WebApplication1/Startup.cs
   :dedent: 8
   :language: c#
   :lines: 20-35
-  :emphasize-lines: 1,6
+  :emphasize-lines: 6
 
 The ``IHostingEnvironment`` service is used to get the current environment. In the ``Development`` environment, the highlighted line of code above would look for a file named ``appsettings.Development.json`` and use its values, overriding any other values, if it's present. Learn more about :doc:`environments`.
 
 When specifying files as configuration sources, you can optionally specify whether changes to the file should result in the settings being reloaded. This is configured by passing in a ``true`` value for the ``reloadOnChange`` parameter when calling ``AddJsonFile`` or similar file-based extension methods.
 
-.. warning:: You should never store passwords or other sensitive data in provider code or in plain text configuration files. You also shouldn't use production secrets in your development or test environments. Instead, such secrets should be specified outside the project tree, so they cannot be accidentally committed into the provider repository. Learn more about :doc:`environments` and managing :doc:`/security/app-secrets`.
+.. warning:: You should never store passwords or other sensitive data in configuration source code or in plain text configuration files. You also shouldn't use production secrets in your development or test environments. Instead, such secrets should be specified outside the project tree, so they cannot be accidentally committed into the configuration source repository. Learn more about :doc:`environments` and managing :doc:`/security/app-secrets`.
 
 One way to leverage the order precedence of ``Configuration`` is to specify default values, which can be overridden. In the console application below, a default value for the ``username`` setting is specified in a ``MemoryConfigurationSource``, but this is overridden if a command line argument for ``username`` is passed to the application. You can see in the output how many different configuration sources are configured in the application at each stage of its execution.
```

Comment:
I am not super keen on mentioning `MemoryConfigurationSource` or the other `IConfigurationSources` if this is supposed to be an intro because they are a building block mostly for provider writers as opposed to consumers of configuration. Most consumers should just call the sugar `AddXyz()` extension method. They belong better in in-depth material.